include/internal/sockets.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 1995-2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+
+#ifndef HEADER_OPENSSL_SOCKETS
+
+# ifdef OPENSSL_NO_SOCK
+# elif defined(WINDOWS) || defined(MSDOS)
```

Comment:
I ```OPENSSL_SYS_``` changes.  I did not add e_os2.h because the four files that need this header already include it :)  Recall that this pulls a huge pile of ifdef and include out of all openssl compilations!